.github/workflows/ci.yml

```diff
@@ -0,0 +1,66 @@
+name: CI
+
+on: [push, pull_request]
```

Comment:
for the `pull_request` hook, the [three default events](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request) enabled are the `opened`, `reopened`, and `synchronize`. 

The `synchronize` event happens when the HEAD gets updated in a PR, like when someone pushes to a branch that has a PR open. This is an overlap with the `push` hook. Consider omitting the `synchronize` event to prevent repeated computation.

```yaml
on:
  push:
  pull_request:
    types: [opened, reopened]

```